tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
     $this->drupalGet($node->toUrl());
-    $this->assertEquals('Registration will open today, 1 February 2020, 16:00.', $registration_info_content->getText());
 
-    // Assert registration date description when registration will end today.
-    $registration_start_date = (clone $static_time)->modify('-10 days');
-    $registration_end_date = (clone $static_time)->modify('+1 hours');
+    $this->assertRegisterButtonEnabled($registration_content);
+    $this->assertEquals('Book your seat, 1 week left to register, registration will end on 28 February 2020, 01:00', $registration_info_content->getText());
+
+    // Assert "Registration date" field when registration will finish today.
+    $registration_start_date = (clone $static_time)->modify('- 10 days');
+    $registration_end_date = (clone $static_time)->modify('+ 1 hours');
     $node->set('oe_event_registration_dates', [
       'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
       'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
     ])->save();
-    $this->setTimezone('Australia/Sydney');
     $this->drupalGet($node->toUrl());
-    $this->assertEquals('Book your seat, the registration will end today, 2 February 2020, 04:00', $registration_info_content->getText());
 
-    $this->setTimezone('America/New_York');
-    \Drupal::entityTypeManager()->getViewBuilder('node')->resetCache([$node]);
+    $this->assertRegisterButtonEnabled($registration_content);
+    $this->assertEquals('Book your seat, the registration will end today, 18 February 2020, 02:00', $registration_info_content->getText());
+
+    // Assert "Registration date" field when registration will start today.
+    $registration_start_date = (clone $static_time)->modify('+ 1 hour');
+    $registration_end_date = (clone $static_time)->modify('+ 10 days');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $this->assertRegisterButtonDisabled($registration_content);
+    $this->assertEquals('Registration will open today, 18 February 2020, 02:00.', $registration_info_content->getText());
+
+    // Assert "Registration date" field when registration will start in future.
+    $registration_start_date = (clone $static_time)->modify('+ 1 day');
+    $registration_end_date = (clone $static_time)->modify('+ 10 days');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
```

Comment:
See above, when testing time-related features we must never change the actual event dates, we should change time, run cron and reload the page, instead.